SocketIOClientSwift/SocketEngine.swift

```diff
@@ -246,63 +246,63 @@ public final class SocketEngine: NSObject, WebSocketDelegate, SocketLogClient {
         
         doRequest(req)
     }
-    
+
     private func doRequest(req:NSMutableURLRequest) {
         if !polling {
             return
         }
-        
+
         req.cachePolicy = NSURLRequestCachePolicy.ReloadIgnoringLocalAndRemoteCacheData
-        
+
         SocketLogger.log("Doing polling request", client: self)
-        
+
         session.dataTaskWithRequest(req) {[weak self] data, res, err in
             if let this = self {
-                if err != nil {
+                if err != nil || data == nil {
                     if this.polling {
-                        this.handlePollingFailed(err.localizedDescription)
+                        this.handlePollingFailed(err?.localizedDescription ?? "Error")
                     } else {
-                        SocketLogger.err(err.localizedDescription, client: this)
+                        SocketLogger.err(err?.localizedDescription ?? "Error", client: this)
                     }
                     return
                 }
-                
+
                 SocketLogger.log("Got polling response", client: this)
-                
-                if let str = NSString(data: data, encoding: NSUTF8StringEncoding) as? String {
+
+                if let str = NSString(data: data!, encoding: NSUTF8StringEncoding) as? String {
                     dispatch_async(this.parseQueue) {[weak this] in
                         this?.parsePollingMessage(str)
                     }
                 }
-                
+
                 this.waitingForPoll = false
-                
+
                 if this.fastUpgrade {
                     this.doFastUpgrade()
                 } else if !this.closed && this.polling {
                     this.doPoll()
                 }
-            }}.resume()
+            }}?.resume()
```

Comment:
You probably don't want to ignore tasks that failed to be created, failing silently. Maybe change this to `!` instead?